Add Agent Manager for multi-agent orchestration

marius-kilocode opened the pull request:
Adds a new Agent Manager panel for running multiple autonomous agent sessions in parallel.

**Features:**
- Adds an Agent Manager button to the kilo code sidebar
- Each agent is driven by a in process isolated kilo code cli binary
- Users can start multiple agent sessions in parallel in the same repository
- Start, stop, and remove sessions
- Session state managed via Jotai in webview

**Notes:**
- The parallel worktree integration is not part of this PR to keep it smaller
- Behind feature flag (`AGENT_MANAGER_ENABLED`) - dev mode only
- Agents run in full-auto mode without user confirmation, but user gets a notification
- Requires kilocode CLI to be installed (we validate this)
- We also validate for empty repositories
- All localizations are added

**Screens**
<img width="297" height="111" alt="image" src="https://github.com/user-attachments/assets/9f9440d2-da98-4ad0-bbbc-8891c2f371b5" />
<img width="781" height="734" alt="image" src="https://github.com/user-attachments/assets/30149732-fedb-487d-a6fd-25ff92c016ca" />
<img width="933" height="617" alt="image" src="https://github.com/user-attachments/assets/e91e284c-7473-4c8f-88b3-ae8aec318dfd" />
<img width="749" height="87" alt="image" src="https://github.com/user-attachments/assets/4178b9f4-fb07-45ef-8d50-b44c7cb5c03c" />

Commit:
feat: add telemetry tracking for autocomplete

This PR adds comprehensive telemetry tracking for inline assist (autocomplete) suggestions to measure the full lifecycle of autocomplete requests.

| Event | Description |
|-------|-------------|
| `AUTOCOMPLETE_SUGGESTION_REQUESTED` | Fired when an autocomplete request is initiated (completion provider invoked by VS Code) |
| `AUTOCOMPLETE_LLM_REQUEST_COMPLETED` | Fired when the LLM request completes successfully (includes latency, cost, tokens) |
| `AUTOCOMPLETE_LLM_REQUEST_FAILED` | Fired when the LLM request fails (includes latency and error message) |
| `AUTOCOMPLETE_LLM_SUGGESTION_RETURNED` | Fired when a newly fetched suggestion is returned to the user |
| `AUTOCOMPLETE_SUGGESTION_CACHE_HIT` | Fired when a suggestion is served from cache |
| `AUTOCOMPLETE_ACCEPT_SUGGESTION` | Fired when a user accepts an inline suggestion |
| `AUTOCOMPLETE_SUGGESTION_FILTERED` | Fired when a suggestion is filtered out (empty response or postprocessing) |

1. **Request Lifecycle Tracking**: Events are captured at each stage of the autocomplete flow:
   - Request initiated → Cache check → LLM request → Response processing → Delivery

2. **Acceptance Tracking**: Registered a VSCode command (`kilocode.ghost.inlineAssist.accepted`) that fires when suggestions are accepted

3. **Filtering Tracking**: Captures when suggestions are filtered with reason:
   - `empty_response`: LLM returned no content
   - `filtered_by_postprocessing`: Suggestion was filtered by quality checks

4. **LLM Request Metrics**: Captures latency, cost, and token usage for completed requests, and latency + error for failed requests

5. **Backward Compatibility**: All telemetry calls are protected with `TelemetryService.hasInstance()` checks

- `packages/types/src/telemetry.ts` - Added new telemetry event types
- `src/services/ghost/classic-auto-complete/AutocompleteTelemetry.ts` - New telemetry helper module
- `src/services/ghost/classic-auto-complete/GhostInlineCompletionProvider.ts` - Integrated telemetry tracking
- `src/services/ghost/GhostServiceManager.ts` - Simplified cost tracking callback
- `src/core/webview/ClineProvider.ts` - Added telemetry for inline assist quick task

- Added test coverage for acceptance tracking via command callback
- All existing tests continue to pass ✅

This telemetry will provide valuable metrics on:
- Suggestion acceptance rates
- Cache hit rates
- LLM request latency and failure rates
- Suggestion quality (filtering rates)
- User engagement with inline assist

iscekic committed:
hijack flow on /session

iscekic committed:
add /session share command

iscekic committed:
add /session fork

iscekic committed:
update share url

iscekic committed:
fix paths

iscekic committed:
improve messaging if share id not present

iscekic committed:
add changesets

marius-kilocode committed:
Add initial agent manager draft implementation

marius-kilocode committed:
Add auto-mode banner and fix CSP

marius-kilocode committed:
Add feature flag

iscekic committed:
add share session button

marius-kilocode committed:
Validate that kilocode cli is installed and link to docs

marius-kilocode committed:
Fix edge cases in cli output handling and avoid dropping final message

marius-kilocode committed:
Use circuit-board icon for agent manager

iscekic committed:
add env var disabling session functionality

useful for integration tests

iscekic committed:
improve command param naming

sessions can be forked from a share or from own session id

marius-kilocode committed:
Remove PID from UI

marius-kilocode committed:
Fix type checks in tests

marius-kilocode committed:
Merge branch 'main' into agent-manager

marius-kilocode committed:
Improve layout, ensure UI validation, add new agent tab, remove duplicate buttons

marius-kilocode committed:
Add localizations for all agent manager texts

iscekic committed:
improve copy

marius-kilocode committed:
Add missing I18nextProvider

iscekic committed:
handle /kilo/fork?id=xxx urls

marius-kilocode committed:
Align font styles

marius-kilocode committed:
Align start agent input field

marius-kilocode committed:
Fix infinite spinner when no folder is selected

iscekic committed:
fix copy

iscekic committed:
add translations

marius-kilocode committed:
Change loading spinner

marius-kilocode committed:
Handle different message events correctly

marius-kilocode committed:
Revert ripgrep fallback

iscekic committed:
fix failing cli test

marius-kilocode committed:
Fix lint issues

marius-kilocode committed:
Add missing translations

marius-kilocode committed:
Fix Copilot nitties

marius-kilocode committed:
Fix bug where sessions could not be deleted

marius-kilocode committed:
Add kilocode markers

marius-kilocode committed:
Update jotai

marius-kilocode committed:
Fix 3 further p2/p3 review issues

marius-kilocode committed:
Update pnpm-lock.yaml

marius-kilocode committed:
Merge branch 'main' into agent-manager

RSO commented on src/core/agent-manager/AgentManagerProvider.ts:
```diff
@@ -0,0 +1,623 @@
+import * as vscode from "vscode"
```
Let's move this file to a `kilocode/` folder to make it clear it's Kilo-only

marius-kilocode replied:
I will move them to the existing kilocode dirs.
I noticed that the existing core logic lives in `src/core/kilocode/` instead of `src/kilocode/core` was there an intentional decision to cut it in core @RSO ?

RSO commented on src/core/agent-manager/AgentManagerProvider.ts:
```diff
@@ -0,0 +1,623 @@
+const SESSION_TIMEOUT_MS = 120_000 // 2 minutes
+const WINDOWS = "win32"
+
+export function getKilocodeCliCandidatePaths(env: NodeJS.ProcessEnv, platform: NodeJS.Platform): string[] {
```
I'm not sure about this implementation. Can't we use the `PATH` to look up `kilocode`? Since I use `mise-en-place` to install NodeJS, my `kilocode` bin is located at `/Users/remon/.local/share/mise/installs/node/22.15.1/bin/kilo`, and from the looks of it I'm not sure if this function will detect that.

RSO commented on src/core/agent-manager/AgentManagerProvider.ts:
```diff
@@ -0,0 +1,623 @@
+			} catch (error) {
+				this.outputChannel.appendLine(`[AgentManager] Error checking CLI path ${path}: ${error}`)
+			}
+		}
```
My worry about logic like this is that it takes up a lot of space in files and is often hard to understand. And since nobody should run through this (because they should have `kilocode` on their PATH), when they do, we don't know that the outcome of the method will be the outcome that we expect.

RSO reviewed (changes requested):
Let's move all the `agent-manager` directories to `kilocode/` subdirectories to prevent potential merge conflicts. Also, I tried to test the Agent Manager, but I got an empty screen:

<img width="3330" height="2164" alt="Image" src="https://github.com/user-attachments/assets/84ee3779-fe02-4aed-b6d5-9d85abd99d15" />

marius-kilocode committed:
Merge branch 'main' into agent-manager

Commit:
--amend

marius-kilocode committed:
Remove kilocode_change annotation from feature-flags

Commit:
add changeset

Commit:
feat(telemetry): add rich metadata to autocomplete telemetry events

Add additional context to autocomplete telemetry events for better analytics:

- Add AutocompleteContext interface with languageId, modelId, provider, strategy
- Add CacheMatchType to track how suggestions are matched (exact/partial_typing/backward_deletion)
- Add suggestionLength to cache hit and LLM suggestion returned events
- Update findMatchingSuggestion to return match type along with suggestion text
- Update all telemetry capture functions to accept context and new metadata

New properties by event:
- AUTOCOMPLETE_SUGGESTION_REQUESTED: languageId, modelId, provider
- AUTOCOMPLETE_SUGGESTION_CACHE_HIT: languageId, modelId, provider, matchType, suggestionLength
- AUTOCOMPLETE_LLM_SUGGESTION_RETURNED: languageId, modelId, provider, strategy, suggestionLength
- AUTOCOMPLETE_LLM_REQUEST_COMPLETED: languageId, modelId, provider, strategy
- AUTOCOMPLETE_LLM_REQUEST_FAILED: languageId, modelId, provider, strategy
- AUTOCOMPLETE_SUGGESTION_FILTERED: languageId, modelId, provider, strategy, reason

marius-kilocode committed:
refactor: extract registerAgentManager

marius-kilocode committed:
Move agent-manager related code to kilocode directories

iscekic committed:
improve title summary request

marius-kilocode committed:
Add HMR support to autoload agent manager during development builds

marius-kilocode committed:
Cleanup AgentRegistry

marius-kilocode committed:
Update src/core/task/Task.ts

Apply kilocode_change marker suggestion

Co-authored-by: Remon Oldenbeuving <r.s.oldenbeuving@gmail.com>

marius-kilocode committed:
Simplify cli path resolver

marius-kilocode committed:
Merge branch 'main' into agent-manager

iscekic committed:
make the singleton safe to use before dependencies are initialized

iscekic committed:
correctly reinit singleton timer after destroying

marius-kilocode committed:
Make agent manager provider tests platform agnostic

iscekic committed:
regenerate tests

iscekic committed:
extract fn for /session command handling

iscekic committed:
rename button component

iscekic committed:
add marker

iscekic committed:
improve usage of SessionManager

iscekic committed:
add back flow interruption when handling session command

marius-kilocode committed:
Ensure that all agents are disposed when agent manager is closed by user

iscekic committed:
extract function for finding kilocodeToken

marius-kilocode committed:
Merge branch 'main' into agent-manager

marius-kilocode committed:
Log ignored catch phrase

marius-kilocode committed:
Cleanup comments

brianc committed:
Fix crash loop if ManagedIndexer somehow doesn't instantiate on extension startup (#4171)

* Fix crash loop if ManagedIndexer somehow doesn't instantiate on extension startup

* Add changeset

* Add telemetry to managedindexer instance being missing

github-actions[bot] committed:
changeset version bump

marius-kilocode committed:
Merge branch 'main' into agent-manager

marius-kilocode committed:
Merge branch 'main' into agent-manager

RSO reviewed (approved):
I tested it, and while there's definitely room for improvement, I think this is a good place to start with a first version. Great job!

marius-kilocode committed:
Remove chilly-signs-bow.md

marius-kilocode committed:
Merge branch 'main' into agent-manager

marius-kilocode committed:
Use test naming convention for test files

marius-kilocode committed:
Rename webview-ui relate test files to test convention
